src/client_side_request.cc

```diff
@@ -437,7 +438,9 @@ clientFollowXForwardedForCheck(Acl::Answer answer, void *data)
                 /* override the default src_addr tested if we have to go deeper than one level into XFF */
                 Filled(calloutContext->acl_checklist)->src_addr = request->indirect_client_addr;
             }
-            calloutContext->acl_checklist->nonBlockingCheck(clientFollowXForwardedForCheck, data);
+            if (++calloutContext->currentXffHopNumber < MAX_X_FORWARDED_HOP) {
+                calloutContext->acl_checklist->nonBlockingCheck(clientFollowXForwardedForCheck, data);
+            }
```

Comment:
No need for `{...}` around single code lines.

```suggestion
            if (++calloutContext->currentXffHopNumber < MAX_X_FORWARDED_HOP)
                calloutContext->acl_checklist->nonBlockingCheck(clientFollowXForwardedForCheck, data);
```

OR, if the `#define MAX_X_FORWARDED_HOP` is not needed to be build-time tuned this could be:
```suggestion
            static const uint32_t maxXffHops = 16;
            if (++calloutContext->currentXffHopNumber < maxXffHops)
                calloutContext->acl_checklist->nonBlockingCheck(clientFollowXForwardedForCheck, data);
```